consensus/types/src/chain_spec.rs

```diff
@@ -607,6 +608,16 @@ impl ChainSpec {
         }
     }
 
+    /// Returns the deneb preset value if peerdas epoch hasn't hit.
+    /// Otherwise, returns the value obtained from the config.yaml.
+    pub fn max_blobs_per_block(&self, epoch: Epoch) -> u64 {
+        if self.is_peer_das_enabled_for_epoch(epoch) {
+            self.max_blobs_per_block
+        } else {
+            default_max_blobs_per_block()
```

Comment:
Is this necessary? If we're changing this on devnets, then all clients would be using  `MAX_BLOBS_PER_BLOCK` from deneb, and we won't be able to interop with other clients. This config value is already configurable for some clients for Deneb, so if we're overriding this value, other clients would be using this value even before PeerDAS.

For PeerDAS purpose I guess we'd be using a new config value when the time comes?

Reply:
I think I see what you're trying to do here - this function returns the `max_blobs_per_block` for the epoch, so in the future it could return `MAX_BLOBS_PER_BLOCK` for deneb, and `MAX_BLOBS_PER_BLOCK_ELECTRA` if we're in Electra?

The issue i see with returning the `default` is that if we start a devnet on `Deneb` with a different `MAX_BLOBS_PER_BLOCK`, we won't be able to interop with other clients (which is the main purpose of making this field configurable)

Reply:
I see your point. However, I wanted to be conservative with making changes that could allow configurability with `max_blobs_per_block` in deneb since that might screw with mainnet.

From what I have seen in devnets, the electra/peerdas activation happens on the first epoch so we can get the configurability from the first epoch onwards. I didn't want to overengineer for a usecase that might not exist on devnets specially because it could lead to consensus issues on mainnet

Reply:
Yeah I agree being conservative on this is a good idea - although I think people testing increased blob count (for both peerdas / without peerdsa) would assume this works by just modifying `max_blobs_per_block` - but this wouldn't work without enabling peerdas. On the other hand, I also think once we test this on mainnet, it should be fine and after that we shouldn't randomly commit a modification to mainnet config.

However, I don't feel strongly about this, just wanted to bring this up and I do really see a good reason to keep it as it is! 👍